dmoj/cli.py

```diff
@@ -202,17 +231,34 @@ def execute(self, line):
                 except Exception as io:
                     err = str(io)
             else:
-                src = []
-                try:
-                    while True:
-                        s = raw_input()
-                        if s.strip() == ':q':
-                            raise EOFError
-                        src.append(s)
-                except EOFError:  # Ctrl+D
-                    src = '\n'.join(src)
-                except Exception as io:
-                    err = str(io)
+                if language_id in ['PY2', 'PYPY2']:
```

Comment:
Are you so brain damaged as to not realize there is `.ext` attribute on the executor? How the hell did you think executors know how to create source files?? Stop reinventing the goddamn wheel in a shitty way.